packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/__tests__/MappingTest.test.tsx

```diff
@@ -0,0 +1,133 @@
+    queryState.setTarget(targetClass);
+    queryState.setGraphFetchTree(
+      getRawGraphFetchTreeData(editorStore, targetClass, mapping),
+    );
+    // populate test data
+    const runtimeState = new MappingExecutionJsonModelConnectionRuntimeState(
+      editorStore,
+      mapping,
+    );
+    runtimeState.setSourceClass(sourceClass);
+    runtimeState.setTestData(JSON.stringify(generatedSourceData, undefined, 2));
+    executionState.setRuntimeState(runtimeState);
+    const testData = JSON.parse(runtimeState.testData) as Record<
+      string,
+      string
+    >;
+    expect(testData.sFirstName).toEqual('myFirstName');
+    expect(testData.sLastName).toEqual('myLastName');
+    executionState.setQueryState(queryState);
+    const expectedTestDataDisplay =
+      '{\n  "sFirstName": "myFirstName",\n  "sLastName": "myLastName"\n}';
+    expect(runtimeState.testData).toBe(expectedTestDataDisplay);
+    const inputData = runtimeState.inputData;
+    const jsonInputData = guaranteeType(inputData, ObjectInputData);
+    expect(jsonInputData.sourceClass.value).toBe(sourceClass);
+    expect(jsonInputData.inputType).toBe(OBJECT_INPUT_TYPE.JSON);
+    expect(jsonInputData.data).toBe(
+      '{"sFirstName":"myFirstName","sLastName":"myLastName"}',
+    );
+    // execute mapping
+    await flowResult(executionState.executeMapping());
+    const expectedResultDisplay =
+      '{\n  "defects": [\n\n  ],\n  "source": {\n    "defects": [\n\n    ],\n    "source": {\n      "number": 1,\n      "record": "{\\"sFirstName\\":\\"myFirstName\\",\\"sLastName\\":\\"myLastName\\"}"\n    },\n    "value": {\n      "sFirstName": "myFirstName",\n      "sLastName": "myLastName"\n    }\n  },\n  "value": {\n    "firstName": "myFirstName",\n    "lastName": "myLastName"\n  }\n}';
+    expect(executionState.executionResultText).toBe(expectedResultDisplay);
+    // promote to test and assert correct input and expected result data
+    await flowResult(executionState.promoteToTest());
+    const mappingTest = mapping.tests[0];
+    expect(mappingTest).toBeTruthy();
+    const testInputData = mappingTest.inputData[0];
+    expect(testInputData).toBeTruthy();
+    const testObjectInputData = guaranteeType(testInputData, ObjectInputData);
+    expect(testObjectInputData.sourceClass.value).toBe(sourceClass);
+    expect(testObjectInputData.inputType).toBe(OBJECT_INPUT_TYPE.JSON);
+    expect(testObjectInputData.data).toBe(
+      '{"sFirstName":"myFirstName","sLastName":"myLastName"}',
+    );
+    const testAssert = guaranteeType(
+      mappingTest.assert,
+      ExpectedOutputMappingTestAssert,
+    );
+    const savedExpectedResult =
+      '{"defects":[],"source":{"defects":[],"source":{"number":1,"record":"{\\"sFirstName\\":\\"myFirstName\\",\\"sLastName\\":\\"myLastName\\"}"},"value":{"sFirstName":"myFirstName","sLastName":"myLastName"}},"value":{"firstName":"myFirstName","lastName":"myLastName"}}';
```

Comment:
instead of this, we can also call `trytoMinify...` as well so you don't have to repeat the string 2 times